lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -99,14 +99,16 @@ def success?(response)
   end
 
   def error_message(response)
+    fqdn = URI::parse(resource[:base_url]).host
+
     explanations = {
-      '400' => 'Something is wrong with the data sent to Foreman server',
+      '400' => "Something is wrong with the data sent to #{fqdn}",
       '401' => 'Often this is caused by invalid Oauth credentials',
       '404' => 'The requested resource was not found',
-      '500' => 'Check /var/log/foreman/production.log on Foreman server for detailed information',
-      '502' => 'The webserver received an invalid response from the backend service. Was Foreman unable to handle the request?',
-      '503' => 'The webserver was unable to reach the backend service. Is foreman.service running?',
-      '504' => 'The webserver timed out waiting for a response from the backend service. Is Foreman under unusually heavy load?'
+      '500' => "Check /var/log/foreman/production.log on #{fqdn} for detailed information",
+      '502' => "The webserver received an invalid response from the backend service. Was foreman.service at #{fqdn} unable to handle the request?",
```

Comment:
I still think `Foreman` is more correct. Technically Foreman is delivered by running `foreman.socket` and `foreman.service`. Either could have a problem.

It's also clearly documented that Foreman is a service. https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html/planning_for_red_hat_satellite/chap-red_hat_satellite-architecture_guide-introduction_to_red_hat_satellite#sect-Red_Hat_Satellite-Architecture_Guide-Red_Hat_Satellite_6_System_Components states:

>Foreman
>    Foreman is an open source application used for provisioning and life cycle management of physical and virtual systems. Foreman automatically configures these systems using various methods, including kickstart and Puppet modules. Foreman also provides historical data for reporting, auditing, and troubleshooting.

Reply:
I would tend to agree. I think the trouble only comes when we combine the term to be 'Foreman server'. If we stick to either Foreman, Foreman service or Foreman application then it works for both cases as it identifier the actual service and not the higher level concept of the entire box that 'Foreman server' or 'Satellite server' that can lead to branding debates.

Reply:
I went with the language `Foreman at #{fqdn}` in my latest update. Please let me know if there's any more feedback on the wording... thanks!